### PR TITLE
Added feature with multiflile configuration in ocelot and swagger

### PR DESCRIPTION
**Pull request by dadjh85**
@Burgyn Here I leave you a pull request of what we have been talking about days ago, sorry for the delay in sending it, but I have been busy for several days. There is a lack of documentation, but if you like the code I will write the documentation because if not, it is not worth to write it.

### CODE REVIEW

**Comment by Burgyn** [comment]
Hi,
thanks for your PR.
I'll look into it tomorrow.

In the meantime, Can you please try to resolve the merge conflicts?

**Comment by dadjh85** [comment]
> 
> 
> Hi,
> thanks for your PR.
> I'll look into it tomorrow.
> 
> In the meantime, can you please try to resolve the merge conflicts?

I can't click the button to resolve the conflicts, the button appears disabled, you may not have permission for this action.

File: demo/ApiGateway/Startup.cs [hunk 1]
```diff
@@ -59,6 +59,8 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
                         new KeyValuePair<string, string>("Key", "Value"),
                         new KeyValuePair<string, string>("Key2", "Value2"),
                     };
+
+                    //opt.ServerOcelot = "/siteName/apigateway" ;
```
Comment (Burgyn) [inline demo/ApiGateway/Startup.cs:63]: Please remove commented code.

File: src/MMLib.SwaggerForOcelot/Configuration/SwaggerFileConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+using System.Collections.Generic;
+
+namespace MMLib.SwaggerForOcelot.Configuration
+{
+    /// <summary>
+    /// Mapping of the files configuration of ocelot
+    /// </summary>
+    public class SwaggerFileConfiguration
+    {
+        public List<SwaggerFileRoute> Routes { get; set; } = new List<SwaggerFileRoute>();
```
Comment (Burgyn) [inline src/MMLib.SwaggerForOcelot/Configuration/SwaggerFileConfiguration.cs:11]: Please comment all public methods.

File: src/MMLib.SwaggerForOcelot/DependencyInjection/ConfigurationBuilderExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace MMLib.SwaggerForOcelot.DependencyInjection
+{
+    public static class ConfigurationBuilderExtensions
```
Comment (Burgyn) [inline src/MMLib.SwaggerForOcelot/DependencyInjection/ConfigurationBuilderExtensions.cs:14]: Please comment public members.

File: src/MMLib.SwaggerForOcelot/DependencyInjection/ConfigurationBuilderExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        /// <param name="folder"></param>
+        /// <param name="fileOfSwaggerEndPoints"></param>
+        /// <returns>a Object IConfigurationBuilder</returns>
+        public static IConfigurationBuilder AddOcelotWithSwaggerSupport(
+            this IConfigurationBuilder builder,
+            IWebHostEnvironment environment = null,
+            string folder = "/",
+            string fileOfSwaggerEndPoints = SwaggerForOcelotFileOptions.SwaggerEndPointsConfigFile)
+        {
+            List <FileInfo> files = GetListOfOcelotFiles(folder, environment?.EnvironmentName);
```
Comment (Burgyn) [inline src/MMLib.SwaggerForOcelot/DependencyInjection/ConfigurationBuilderExtensions.cs:30]:
```suggestion
            List<FileInfo> files = GetListOfOcelotFiles(folder, environment?.EnvironmentName);
```

File: src/MMLib.SwaggerForOcelot/DependencyInjection/ConfigurationBuilderExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+        private static List<FileInfo> GetListOfOcelotFiles(string folder, string nameEnvirotment)
+        {
+            string subConfigPattern = @"^ocelot\.(.*?)\.json$";
+
+            var reg = new Regex(subConfigPattern, RegexOptions.IgnoreCase | RegexOptions.Singleline);
+
+            IEnumerable<FileInfo> ocelotFiles = new DirectoryInfo(folder).EnumerateFiles()
+                                                                         .Where(fi => reg.IsMatch(fi.Name));
+            if (!nameEnvirotment.IsNullOrWhiteSpace())
+                ocelotFiles = ocelotFiles.Where(fi => fi.Name.Contains(nameEnvirotment));
```
Comment (Burgyn) [inline src/MMLib.SwaggerForOcelot/DependencyInjection/ConfigurationBuilderExtensions.cs:59]: Wrap if block with `{}` parentheses.

File: src/MMLib.SwaggerForOcelot/DependencyInjection/ConfigurationBuilderExtensions.cs [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+using System.Text.RegularExpressions;
+
+namespace MMLib.SwaggerForOcelot.DependencyInjection
+{
+    public static class ConfigurationBuilderExtensions
+    {
+        /// <summary>
+        /// Extension for compatibility of functionality multifile of ocelot
+        /// </summary>
+        /// <param name="builder"></param>
```
Comment (Burgyn) [inline src/MMLib.SwaggerForOcelot/DependencyInjection/ConfigurationBuilderExtensions.cs:19]: Please write comments for all prameters.

File: src/MMLib.SwaggerForOcelot/DependencyInjection/ConfigurationBuilderExtensions.cs [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+        /// <param name="environment"></param>
+        /// <param name="folder"></param>
+        /// <param name="fileOfSwaggerEndPoints"></param>
+        /// <returns>a Object IConfigurationBuilder</returns>
+        public static IConfigurationBuilder AddOcelotWithSwaggerSupport(
```
Comment (Burgyn) [inline src/MMLib.SwaggerForOcelot/DependencyInjection/ConfigurationBuilderExtensions.cs:24]: Please write unit test for this method.

**Comment by Burgyn** [comment]
> > Hi,
> > thanks for your PR.
> > I'll look into it tomorrow.
> > In the meantime, can you please try to resolve the merge conflicts?
> 
> I can't click the button to resolve the conflicts, the button appears disabled, you may not have permission for this action. i read this text: "Only those with write access to this repository can merge pull requests."

Hi,
yes, button is disabled. But you can merge master branch to your branch and resolve conflicts. Thanks.

Code looks good, I wrote some issues directly to code. Please check it.
Thanks a lot.

**Comment by dadjh85** [comment]
> 
> 
> > > Hi,
> > > thanks for your PR.
> > > I'll look into it tomorrow.
> > > In the meantime, can you please try to resolve the merge conflicts?
> > 
> > 
> > I can't click the button to resolve the conflicts, the button appears disabled, you may not have permission for this action. i read this text: "Only those with write access to this repository can merge pull requests."
> 
> Hi,
> yes, button is disabled. But you can merge master branch to your branch and resolve conflicts. Thanks.
> 
> Code looks good, I wrote some issues directly to code. Please check it.
> Thanks a lot.

Ok, this week, I'll review the changes and update this pull request.

**Comment by dadjh85** [comment]
Hi @Burgyn, I'm updating this pull request with the changes you mentioned and also I've updated the demo project to work with multiple swagger files inside the configuration folder, I've added the different ocelot.SERVICE.json files, I don't have much experience with this kind of tests focused on components, but I've been checking over and maybe I can try to implement these tests, anyway if I have any complication or doubt I'll tell you if you help me to complete or finish them, otherwise you can check the update of this pull request

**Comment by Burgyn** [comment]
Hi,

thanks. I will write unit tests later.

One unit test began to fall. Will you look at it?
[ReturnUriFromServiceDiscoveryWhenRouteDoesntExist ](https://github.com/Burgyn/MMLib.SwaggerForOcelot/pull/109/checks?check_run_id=773300238)

**Comment by dadjh85** [comment]
> Hi,
> 
> thanks. I will write unit tests later.
> 
> One unit test began to fall. Will you look at it?

OK, fine, I'll try and look at it today, the test that fails, by tomorrow at the latest I should have it sorted out.

**Comment by dadjh85** [comment]
> Hi,
> 
> thanks. I will write unit tests later.
> 
> One unit test began to fall. Will you look at it?
> [ReturnUriFromServiceDiscoveryWhenRouteDoesntExist ](https://github.com/Burgyn/MMLib.SwaggerForOcelot/pull/109/checks?check_run_id=773300238)

Hi, I've already fixed the test that failed and I've also written the documentation about the developed feature

**Comment by Burgyn** [comment]
Hi,

thanks a lot. This feature will be available in version `2.4.0` soon.

**Pull request merged**